### DISCUSSION TITLE
Images not loading when using Glide tag

### DISCUSSION THREAD

**Discussion by st1pps** [statamic/cms#11146] [Q&A]
Hi, I'm just starting to experiment with Statamic, and saw that there was an earlier discussing with my problem that did not get resolved: https://github.com/statamic/cms/discussions/7550

PHP 8.2.24
Statamic 5.38.1
I'm using DDEV to locally host the page.

I've got some global assets fields with images. When I render them directly in my antlers template, they display just fine:

```
<img src="{{ theme:logo }}" />
``` 
becomes: 
```
<img src="/assets/logophysio1-rossmann.jpg">
```

In that case the image displays just fine. When I use Glide:

```
{{ glide :src="theme:logo"  }}
     <img src="{{ url }}"  />
{{ /glide }}
```
this gets rendered:
```
<img src="/img/asset/YXNzZXRzL2xvZ29waHlzaW8xLXJvc3NtYW5uLmpwZw==/logophysio1-rossmann.jpg?s=20379fb8d6fefceb3b0748a3f351d554" >
```
But no image is shown. When I open the image link directly in the browser, the following error is displayed:

Symfony \Component \HttpKernel \Exception\HttpException
Signature is missing.

Any idea what might be going wrong? I have just added content so far, and not customized any of the standard statamic settings.

**Comment by jasonvarga** [comment]
When you open the image directly in the browser, do you still have the `?s=203...` part in the url?

**Reply by st1pps** [reply]
Yes, I have copied the complete content from the src attribute into the browser bar.

**Reply by jasonvarga** [reply]
It seems like the query string is being ignored by your server. Maybe something DDEV is doing. 🤔  I don't really know anything about DDEV but if you can provide the nginx config file it uses, maybe we can help.

**Reply by st1pps** [reply]
The problem does seem to be DDEV, thank you! After some problems I've got it running with Herd just to give it a try, under which Glide worked.

Here is the nginx conf. It does try to import more configs at the bottom, but I inspected the mounted drive inside the docker container, there were no additional config files.

```
# ddev laravel config

#ddev-generated
# If you want to take over this file and customize it, remove the line above
# and ddev will respect it and won't overwrite the file.
# See https://ddev.readthedocs.io/en/stable/users/extend/customization-extendibility/#custom-nginx-configuration

server {
    listen 80 default_server;
    listen 443 ssl default_server;

    root /var/www/html/public;

    ssl_certificate /etc/ssl/certs/master.crt;
    ssl_certificate_key /etc/ssl/certs/master.key;

    include /etc/nginx/monitoring.conf;

    index index.html index.htm index.php;

    charset utf-8;

    # Disable sendfile as per https://docs.vagrantup.com/v2/synced-folders/virtualbox.html
    sendfile off;
    error_log /dev/stdout info;
    access_log /var/log/nginx/access.log;

    location / {
        absolute_redirect off;
        try_files $uri $uri/ /index.php?$query_string;
    }

    location = /favicon.ico { access_log off; log_not_found off; }
    location = /robots.txt  { access_log off; log_not_found off; }

    error_page 404 /index.php;

    location ~ \.php$ {
        fastcgi_split_path_info ^(.+\.php)(/.+)$;
        fastcgi_pass unix:/run/php-fpm.sock;
        fastcgi_buffers 16 16k;
        fastcgi_buffer_size 32k;
        fastcgi_param SCRIPT_FILENAME $document_root$fastcgi_script_name;
        fastcgi_param SCRIPT_NAME $fastcgi_script_name;
        fastcgi_index index.php;
        include fastcgi_params;
        fastcgi_intercept_errors off;
        # fastcgi_read_timeout should match max_execution_time in php.ini
        fastcgi_read_timeout 10m;
        fastcgi_param SERVER_NAME $host;
        fastcgi_param HTTPS $fcgi_https;
        # Pass the X-Accel-* headers to facilitate testing.
        fastcgi_pass_header "X-Accel-Buffering";
        fastcgi_pass_header "X-Accel-Charset";
        fastcgi_pass_header "X-Accel-Expires";
        fastcgi_pass_header "X-Accel-Limit-Rate";
        fastcgi_pass_header "X-Accel-Redirect";
    }

    # Expire rules for static content

    # Media: images, icons, video, audio, HTC
    location ~* \.(?:jpg|jpeg|gif|png|ico|cur|gz|svg|svgz|mp4|ogg|ogv|webm|webp|htc)$ {
        expires 1M;
        access_log off;
        add_header Cache-Control "public";
    }

    # Prevent clients from accessing hidden files (starting with a dot)
    # This is particularly important if you store .htpasswd files in the site hierarchy
    # Access to `/.well-known/` is allowed.
    # https://www.mnot.net/blog/2010/04/07/well-known
    # https://tools.ietf.org/html/rfc5785
    location ~* /\.(?!well-known\/) {
        deny all;
    }

    # Prevent clients from accessing to backup/config/source files
    location ~* (?:\.(?:bak|conf|dist|fla|in[ci]|log|psd|sh|sql|sw[op])|~)$ {
        deny all;
    }
    include /etc/nginx/common.d/*.conf;
    include /mnt/ddev_config/nginx/*.conf;
}
```

**Reply by jasonvarga** [reply]
I believe it's this chunk:

```
    location ~* \.(?:jpg|jpeg|gif|png|ico|cur|gz|svg|svgz|mp4|ogg|ogv|webm|webp|htc)$ {
        expires 1M;
        access_log off;
        add_header Cache-Control "public";
    }
```

I'm curious if it works if you comment this section out.

**Reply by st1pps** [reply]
You are right! When I comment out that part of the nginx config the images get served just fine. For local development removing the cache control is fine, but I wonder how that would need to be configured for production. Either way, I have opened a discussion on the DDEV repo, as this is something that should be included in the documentation for how to get startet with DDEV and Statamic (https://github.com/orgs/ddev/discussions/6751). 
Thank you :)

**Reply by jasonvarga** [reply]
The regex can be changed to avoid it on Glide's `img` route.

See https://github.com/statamic/cms/issues/10053#issuecomment-2107981429

To be honest I never understood why this results in a 404 though.
